Buttplug.Core/ButtplugMessage.cs

```diff
@@ -1,12 +1,64 @@
-﻿using Newtonsoft.Json;
+﻿using System;
+using Newtonsoft.Json;
 
 namespace Buttplug.Core
 {
     public class ButtplugMessage
     {
+        /*
+         * Message versions
+         *
+         * These are here for backwards compatibility support, but
+         * this also serves as a changelog of sorts.
+         *
+         * Version 0 - Schema 0.1.0
+         *   First release with no backwards compatibility
+         *
+         * Version 1 - Schema 0.2.0
+         *   Introduction of MessageVersioning
+         *   Addition of generic VibrateCmd
+         */
+        [JsonIgnore]
+        public const uint CurrentMessageVersion = 1;
+
         [JsonProperty(Required = Required.Always)]
         public uint Id { get; set; }
 
+        [JsonIgnore]
+        public uint MessageVersioningVersion
+        {
+            get
+            {
+                return _messageVersioningVersion;
+            }
+
+            protected set
+            {
+                _messageVersioningVersion = value;
+            }
+        }
+
+        [JsonIgnore]
+        public Type MessageVersioningPrevious
```

Comment:
If the above becomes MessageVersion, this should be PreviousMessageVersionType to denote that it's going to be a type, not a version.

Reply:
Enterprise grade is the only world I know, so of course I'm going to aim for enterprise buttplug control!